### PR TITLE
Update CI test report format from xml to csv

### PR DESCRIPTION
**Pull request by chuneuny-emily**
### Summary
The current CI system performs e2e and regression tests, and the test reports are in XML format with columns that require manual interpretation. To simplify the interpretation and make it easier to analyze the results, we have made changes to convert the reports into a CSV format that shows the pass or fail status for each test case.

**Before**
image

**After**
image

Test Results from CI: https://github.com/openvinotoolkit/training_extensions/actions/runs/5331211012

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by jaegukhyun — APPROVED** [review]

File: requirements/dev.txt [hunk 1]
```diff
@@ -8,4 +8,5 @@ coverage
 pytest-timeout
 pytest-mock
 onnx==1.13.0
-onnxruntime==1.14.1
\ No newline at end of file
+onnxruntime==1.14.1
```
Comment (sungmanc) [inline requirements/dev.txt:11]: Is this package needed?

Comment (chuneuny-emily) [reply requirements/dev.txt:11]: > Is this package needed?

This is the original content that was in requirements/dev.txt. The change I made was to add pytest-csv.
image

**Comment by yunchu** [comment]
@chuneuny-emily My small concerns are on the module `pytest-csv`. as you can see in the github (https://github.com/nicoulaj/pytest-csv/issues) for that the latest follow-ups was on more than two years for the issues.

**Comment by chuneuny-emily** [comment]
> My small concerns are on the module `pytest-csv`. as you can see in the github (https://github.com/nicoulaj/pytest-csv/issues) for that the latest follow-ups was on more than two years for the issues.

Yes, it seems that there haven't been any updates to the GitHub code for about two years. For our current testing, I haven't encountered any issues. If any problems arise in the future, I will fix them using alternative methods.

**Review by sungmanc — APPROVED** [review]

**Pull request merged**